controllers/rsyncnodedatamovement_controller.go

```diff
@@ -90,9 +100,55 @@ func (r *RsyncNodeDataMovementReconciler) Reconcile(ctx context.Context, req ctr
 			}
 
 			r.deleteCompletion(rsyncNode.Name)
+			r.deleteCancelCtx(rsyncNode.Name)
+
+			return ctrl.Result{}, nil
```

Comment:
We're in a resource teardown state here, right?  Then this 'return' should be moved outside the if-block.

Reply:
Perhaps the real question is what do we go if found isn't true. Right now we're not doing anything. Do we want to return `ctrl.Result{}, nil` in that case as well?

Reply:
If we get here, then k8s is deleting the resource...we can't expect it to stick around for more work.  In fact, you may want to add a finalizer so that k8s doesn't clobber this resource before it comes back here to let you cleanup that context.

Reply:
This block is handling the case where we've had a completed data movement but for some reason were not able to record the completion.  If we have any failures in recording the completion, it gets stored in the completion map. This means that next time around we check to see if it failed on the update and we try the update again. As @NateRoiger pointed out, with the RetryOnConflict below, this case shouldn't happen much at all but it could happen.

We're not actually deleting the resource in this case, but we are removing the entry from the completion map and the context map b/c there's nothing left to do on the data movement side. However, we're not actually deleting the resource - just marking it as complete. We only want to return if that update was successful.

Reply:
I still think that not returning from here is confusing.

Reply:
If the pod crashes we'd lose the completion map and `found` could be false. I think retrying the data-movement is appropriate in that case.

Reply:
We decided to remove this last attempt retry block from the top of the reconciler loop. We're going to lean on the RetryOnConflict function in the goroutine.